src/protojure/protobuf/serdes/core.clj

```diff
@@ -101,7 +101,6 @@
   (when (some? item)
     (let [data (->pb item)
           len (count data)]
-      (when-not (zero? len)
-        (.writeTag os tag 2);; embedded messages are always type=2 (string)
-        (.writeUInt32NoTag os len)
-        (.writeRawBytes os (bytes data))))))
\ No newline at end of file
+      (.writeTag os tag 2);; embedded messages are always type=2 (string)
```

Comment:
Are there circumstances where some embedded types may want the (zero?) optimization?  My high level thought is I am wondering if this should be conditional, and have certain types (e.g. oneof) disable the optimization.

Reply:
My first thought back when I first encountered this was to have a separate `write-embedded` for `oneof` for this reason, but according to the `proto3` spec this optimization is always semantically incorrect (and therefore means if you're talking to a different protobuf implementation—e.g. in a different language—you could roundtrip a message and have them receive something different than they sent) so I opted for the more general solution.

Reply:
That seems reasonable.  Worst case is, we are putting more bytes on the wire than we need to, which is the more conservative thing to do it seems.